src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -1871,5 +1865,115 @@ private void InterpretBinaryOperation(ILOpcode opcode)
                     break;
             }
         }
+
+        private void InterpretBranch(ref ILReader reader, ILOpcode opcode, int target)
+        {
+            switch (opcode)
+            {
+                case ILOpcode.br_s:
+                case ILOpcode.br:
+                    reader.Seek(target);
+                    break;
+                case ILOpcode.brfalse_s:
+                case ILOpcode.brfalse:
+                    {
+                        int value = PopWithValidation().AsInt32();
```

Comment:
brfase argument can be anything - int32, int64, object reference, managed pointer, unmanaged pointer or native int.

Reply:
Seems `brtrue` can only be either a `native int` or `object reference` (https://docs.microsoft.com/en-us/dotnet/api/system.reflection.emit.opcodes.brtrue?view=netcore-2.2). Will update accordingly

Reply:
http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-335.pdf is the source of truth for IL instructions. It says that all int32, int64, object reference, managed pointer, unmanaged pointer or native int are allowed.

Reply:
Ah ok - the specs for brfalse and brtrue are different

Reply:
Although the spec says `brtrue` takes only a `native int` or `object reference`, in practice I've seen it work with an `int32` which makes me wonder why it shouldn't support `int64`. Case in point: https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgBgATIIwG4CwAUMgMyYBM6AwugN6HoOanIAs6AsgBQCUt9jA/gwC+Q9GJKYsANnRgAdsHQARAPYAZVaoAOPMXQICBCpQCd0AXnSp8howwBmq85xNzL1nO4A86LKhs5eHhec3grMFt7BmQAdnRTKMZRAmEgA===